src/Strategy/SwiftPM.hs

```diff
@@ -30,9 +30,15 @@ discover dir = context "Swift" $ do
 
 findProjects :: (Has ReadFS sig m, Has Diagnostics sig m) => Path Abs Dir -> m [SwiftPackageProject]
 findProjects = walk' $ \dir _ files -> do
-  case findFileNamed "Package.swift" files of
-    Nothing -> pure ([], WalkContinue)
-    Just file -> pure ([SwiftPackageProject file dir Nothing], WalkContinue)
+  let swiftPackageManifestFile = findFileNamed "Package.swift" files
+  let swiftPackageResolvedFile = findFileNamed "Package.resolved" files
+
+  case (swiftPackageManifestFile, swiftPackageResolvedFile) of
+    (Just manifestFile, Just resolvedFile) -> pure ([SwiftPackageProject manifestFile dir (Just resolvedFile)], WalkSkipSome [".build"])
+    (Just manifestFile, Nothing) -> pure ([SwiftPackageProject manifestFile dir Nothing], WalkSkipSome [".build"])
+    -- Package.resolved without Package.swift is not Swift Package Project
```

Comment:
Why is that?

Reply:
It is not a semantically swift package project, without `Package.swift`. It's similar to that of ~ package.json/package.swift and yarn.lock/package.resolved. 

This reference, although does not explicitly states this, but strongly implies that without `Package.swift`, it is not a swift package project. https://github.com/apple/swift-package-manager/tree/main/Documentation#about-packages